ws-server/src/server.rs

```diff
@@ -648,7 +647,7 @@ pub struct Builder<M = ()> {
 	settings: Settings,
 	resources: Resources,
 	middleware: M,
-	id_provider: Arc<dyn IdProvider>,
+	id_provider: Box<dyn IdProvider>,
```

Comment:
So, when the `id_provider` was `Arc`, cloning it just incremented the pointer count and handed out a new pointer. Now, Cloning actually clones the underlying `IdProvider` thing (and allocates a new box each time).

Are we happy with that allocation-on-clone over an increment-atomic-integer-on-clone? I'd have to peruse the code a little more to get a feeling for how often this was cloned, but perhaps the Arc works out better?

Reply:
actually we could kill the `Arc` too just if we require `trait IdProvider: Clone` which I believe is reasonable.

it will be cloned once for every connection, should be ok I guess?!

Reply:
yeah I think so; better than cloning the Box every time :)